### ISSUE TITLE
Method 'get_QueryProvider' in type 'Microsoft.EntityFrameworkCore.Storage.Internal.RelationalDatabaseFacadeDependencie does not exist

### ISSUE THREAD

**Issue by ssteiner** [dotnet/efcore#30011]
I tried updating my solutions from EFCore6 to EFCore7. It's two big projects, using the same DBL lib. Only the DBL lib has EFCore dependencies.

Everything worked out until I ran into an known issue with EFCore.BulkExtensions, which apparently doesn't support EFCore7.

So I reverted. Now all of a sudden, in one of my two projects, I'm getting the `TypeLoadException` mentioned in the title. 

The full exception being 

System.TypeLoadException: 'Method 'get_QueryProvider' in type 'Microsoft.EntityFrameworkCore.Storage.Internal.RelationalDatabaseFacadeDependencies' from assembly 'Microsoft.EntityFrameworkCore.Relational, Version=6.0.10.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' does not have an implementation.'

Here's the code that is being run (exact same code that works in the other project, with the exact same connection string):

```C#
using var db = new AudmDatabaseContext(connectionString);
await db.Database.OpenConnectionAsync(token).ConfigureAwait(false)
```

The connection string looks like this:

Data Source=.\SQLEXpress;Initial Catalog=AUDM_Cisco12;Integrated Security=True;Multiple Active Result Sets=True;Encrypt=False

My `AudmDatabaseContext` extends `DbContext` and overrides `OnConfiguring` as follows

```C#
protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
{
	optionsBuilder.UseSqlServer(connectionString, options => options.EnableRetryOnFailure());
	optionsBuilder.UseLazyLoadingProxies();
	optionsBuilder.EnableSensitiveDataLogging();
	optionsBuilder.ConfigureWarnings(act => act.Ignore(CoreEventId.DetachedLazyLoadingWarning));
	base.OnConfiguring(optionsBuilder);
}
```

Here's the package references for the DB Lib Project

```xml
<ItemGroup>
  <PackageReference Include="AutoMapper" Version="12.0.0" />
  <PackageReference Include="AutoMapper.Collection.EntityFrameworkCore" Version="9.0.0" />
  <PackageReference Include="Newtonsoft.Json" Version="13.0.2" />
  <PackageReference Include="NinjaNye.SearchExtensions" Version="3.0.1" />
  <PackageReference Include="System.ComponentModel.Annotations" Version="5.0.0" />
  <PackageReference Include="System.Configuration.ConfigurationManager" Version="7.0.0" />
  <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="7.0.0" />
  <PackageReference Include="System.Linq.Dynamic.Core" Version="1.2.24" />
<PackageReference Include="EFCore.BulkExtensions" Version="6.5.6" />
<PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.10">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
</PackageReference>
<PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="6.0.10" />
<PackageReference Include="Microsoft.EntityFrameworkCore.Relational">
  <Version>6.0.10</Version>
</PackageReference>
<PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.10" />
<PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="6.0.10">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
</PackageReference>
</ItemGroup>
```

Any idea what could be going on here?

**Comment by roji** [comment]
@ssteiner there's probably some lingering build stuff. I'd advise doing a thorough cleaning (`git clean -xfd`) in your solution directory and rebuilding from scratch.

**Comment by ssteiner** [comment]
@roji : that was my initial guess too, so I cleaned the entire two solutions and rebuilt them after downgrading the EFCore packages. But, I rechecked after your message, manually going through each folder containing any of the solution .csproj files. And while doing so, looking at the csproj files again, I found that one had a reference to Microsoft.EntityFrameworkCore that VS's Nuget Package manager didn't list. And that one was still on version 7. So that's what causes the issue.
